Added header and C++ files for the PrimaryGeneratorAction and DetectorConstruction

dldyment opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1)
<!-- Reviewable:end -->

dldyment committed:
Add files via upload

jacobdeery reviewed (commented):
Cool! A few general comments:

1. It's a bit difficult to review this without any context as to what the code is supposed to be doing. Could you add a few sentences in the PR description to explain what these files are for?
2. Do we have any way of testing that A) This code builds, and B) The code does what it's supposed to do?
3. The variable naming scheme isn't super consistent throughout: I see `fScoringVolume`, `world_sizeZ`, `solidWorld`, and `logicwb`, which is 4 different conventions. Code will always be a bit easier to read if you adopt one convention and stick with it.

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-:-MGsY_7MAmsNNWST9Hym:b2x0zqd)* status: 0 of 4 files reviewed, 15 unresolved discussions (waiting on @dldyment, @jacobdeery, @kylemtam, and @pauladam316)

___
*[BTDetectorConstruction.hh, line 5 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsYedDAzsdWlsTrNEr:-MGsYedDAzsdWlsTrNEs:b-1t494g) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.hh#L5)):*
> ```
> #ifndef BTDetectorConstruction_h
> 
> #define BTDetectorContruction_h 1
> ```
consider just using `#pragma once` instead of these header guards
___
*[BTDetectorConstruction.hh, line 11 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZ-TJ8XQfs3LQ715O:-MGsZ-TJ8XQfs3LQ715P:b-vbgoyd) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.hh#L11)):*
> ```
> class G4VPhysicalVolume;
> 
> class G4LogicalVolume;
> ```
Where do these classes actually get defined? Are they defined in `G4LogicalVolume.hh`? If so, can we just include that header here instead of using a forward declaration?
___
*[BTDetectorConstruction.cpp, line 13 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsYr2n96298ERqNF9k:-MGsYr2n96298ERqNF9l:be0aeal) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L13)):*
> ```
> //....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
> ```
Were these just copied from the examples? I don't think we want them
___
*[BTDetectorConstruction.cpp, line 17 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGs_7Px2ctCyIvviFCe:-MGs_7Px2ctCyIvviFCf:b-y3qzpa) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L17)):*
> ```
> 0
> ```
prefer `nullptr` rather than `0` - makes it clear what the type is
___
*[BTDetectorConstruction.cpp, line 23 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsYvmS5hDCLPbDlOg3:-MGsYvmS5hDCLPbDlOg4:b7tlxz8) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L23)):*
> ```
> BTDetectorConstruction::~BTDetectorConstruction()
> 
> {}
> ```
I don't think we need to define the destructor if it doesn't do anything
___
*[BTDetectorConstruction.cpp, line 32 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZeJT0a-D1zwnvWib:-MGsZeJT0a-D1zwnvWic:b-jzl624) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L32)):*
> ```cpp
> 	G4NistManager* nist = G4NistManager::Instance();
> 
> 	//World Parameters
> ```

None of these change, right? Can we flag them as `const`?
___
*[BTDetectorConstruction.cpp, line 33 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZUCw6jDqbIyueEhX:-MGsZUCw6jDqbIyueEhY:byzgr78) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L33)):*
> ```
> world_sizeXY
> ```
prefer making separate variables for X and Y so that it's easy to change them independently
___
*[BTDetectorConstruction.cpp, line 39 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZalOCtzBkFU3JYGC:-MGsZalOCtzBkFU3JYGD:bfzhu8v) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L39)):*
> ```
> 1 * world_sizeXY
> ```
why do we need the `1 *` here?
___
*[BTDetectorConstruction.cpp, line 76 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZoQ18_90YYHuGcbx:-MGsZoQ18_90YYHuGcby:b-xsanx3) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L76)):*
<details><summary><i alt="summary">Quoted 8 lines of code…</i></summary>

> ```
>     new G4PVPlacement(0,         //no rotation
>         G4ThreeVector(),         //at origin
>         logicwb,                 //logical volume
>         "Water Box",             //name
>         logicWorld,              //mother volume
>         false,                   //no boolean operation
>         0,                       //copy number
>         checkOverlaps);          //check for any overlaps
> ```
</details>

The result of this `new` operation isn't assigned to anything - is this intentional?
___
*[BTPrimaryGeneratorAction.hh, line 5 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsaWI9BOTM9U_JH0Aa:-MGsaWI9BOTM9U_JH0Ab:b-rsf0nz) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.hh#L5)):*
> ```
> #ifndef BTPrimaryGeneratorAction_h
> 
> #define BTPrimaryGeneratorAction_h 1
> ```
`#pragma once`
___
*[BTPrimaryGeneratorAction.hh, line 13 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGs_evvEMSA9c37k3dQ:-MGs_evvEMSA9c37k3dR:b5ppvlw) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.hh#L13)):*
<details><summary><i alt="summary">Quoted 5 lines of code…</i></summary>

> ```
> class G4ParticleGun;
> 
> class G4Event;
> 
> class G4Box;
> ```
</details>

again, let's try to avoid forward declarations
___
*[BTPrimaryGeneratorAction.cpp, line 21 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGs_o_Z0QQzymmHFPuf:-MGs_o_Z0QQzymmHFPug:b-y0uduu) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.cpp#L21)):*
> ```
>     fParticleGun(0),
> 
>     fEnvelopeBox(0)
> ```
same as above, prefer `nullptr` over `0`
___
*[BTPrimaryGeneratorAction.cpp, line 74 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsaByO2BiD7uF5G2PO:-MGsaByO2BiD7uF5G2PP:b-3mn1xa) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.cpp#L74)):*
> ```cpp
>         G4Exception("BTPrimaryGeneratorAction::GeneratePrimaries()",
>             "MyCode0002", JustWarning, msg);
>     }
> ```

The logic here can be simplified by moving the code from the `if (!fEnvelopeBox)` block into this `else` block
___
*[BTPrimaryGeneratorAction.cpp, line 76 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGscHFT-lqoU-QjOcTv:-MGscHFT-lqoU-QjOcTw:b-z3gxnj) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.cpp#L76)):*
> ```
> G4double size = 0.8;
> ```
Why do we scale by 0.8?
___
*[BTPrimaryGeneratorAction.cpp, line 79 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsaT0t40pcuYYtYq-n:-MGsaT0u4k7qOLsmbYWN:b-7ok7q6) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.cpp#L79)):*
<details><summary><i alt="summary">Quoted 7 lines of code…</i></summary>

> ```
> G4double size = 0.8;
> 
>     G4double x0 = size * envSizeXY * (G4UniformRand() - 0.5);
> 
>     G4double yo = size * envSizeXY * (G4UniformRand() - 0.5);
> 
>     G4double z0 = -0.5 * envSizeZ;
> ```
</details>

these can all be `const`


<!-- Sent from Reviewable.io -->

Dhruv581 committed:
Create BTActionInitialzation.cc

Dhruv581 committed:
Create BTActionInitialzation.hh

dldyment committed:
Update BTDetectorConstruction.cpp

dldyment committed:
Update BTDetectorConstruction.hh

dldyment committed:
Update BTActionInitialzation.hh

dldyment committed:
Update BTPrimaryGeneratorAction.cpp

Defines the types of particles that can be used in the application. In this case, there's only 1: an 18 MeV gamma.

dldyment committed:
Update BTPrimaryGeneratorAction.hh

dldyment committed:
Added RunAction and SteppingAction

RunAction: defines one "run" in the simulation. defines dose, calculates it, and displays it add the end of the run

SteppingAction: defines the process in one "step" of the program, i.e. firing a particle, figuring out if it reaches the volume for which dose is being measured, and if yes adding the energy deposit to the total

dldyment commented:
Thanks Jacob!

dldyment commented:
Thanks Jacob!
To respond to a few things (first part is repeating what I said in #payload):

1. As far as I can tell, there's no way to test anything until the entire application is ready to be build. 
2. The variable naming scheme mimics the examples that come with the G4 install, I figured this would be the easiest way for everyone to be consistent since what we write is heavily based off of them. 
3. I copied the oooo00000oooo000 thing because I thought it was fun, but if it's unprofessional/annoying I'm fine with taking it out. I've left them out of the new files I added. 
4. G4 codebases have a habit of defining things like the desturctor that seemingly have no purpose. I'll comment them out, since I agree it should make sense, but they might be needed for some reason. I couldn't fins anything in the documentation.

dldyment reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1)* status: 0 of 10 files reviewed, 15 unresolved discussions (waiting on @dldyment, @jacobdeery, @kylemtam, and @pauladam316)

___
*[BTDetectorConstruction.hh, line 5 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsYedDAzsdWlsTrNEr:-MHPBTmQ46xFuiLHChH_:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.hh#L5)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> #ifndef BTDetectorConstruction_h
> 
> #define BTDetectorContruction_h 1
> ```
consider just using `#pragma once` instead of these header guards
</blockquote></details>

done
___
*[BTDetectorConstruction.cpp, line 17 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGs_7Px2ctCyIvviFCe:-MHPCUzn7iSL-amqod27:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L17)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> 0
> ```
prefer `nullptr` rather than `0` - makes it clear what the type is
</blockquote></details>

done
___
*[BTDetectorConstruction.cpp, line 33 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZUCw6jDqbIyueEhX:-MHPC_a61ULeuv8PhAD1:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L33)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> world_sizeXY
> ```
prefer making separate variables for X and Y so that it's easy to change them independently
</blockquote></details>

done
___
*[BTDetectorConstruction.cpp, line 39 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZalOCtzBkFU3JYGC:-MHPCaY16c6r7Q26ezJi:ba57d7y) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L39)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> 1 * world_sizeXY
> ```
why do we need the `1 *` here?
</blockquote></details>

we don't, I've taken it out
___
*[BTDetectorConstruction.cpp, line 76 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZoQ18_90YYHuGcbx:-MHPCtFn6rnx8M28C46Y:beegr2s) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L76)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
>     new G4PVPlacement(0,         //no rotation
>         G4ThreeVector(),         //at origin
>         logicwb,                 //logical volume
>         "Water Box",             //name
>         logicWorld,              //mother volume
>         false,                   //no boolean operation
>         0,                       //copy number
>         checkOverlaps);          //check for any overlaps
> ```
The result of this `new` operation isn't assigned to anything - is this intentional?
</blockquote></details>

this is another one of those things that G4 examples do that I don't completely understand. my assumption would be that it is intentional, but I'm afraid I don't have a solid answer rn. I'm looking into it. 
___
*[BTPrimaryGeneratorAction.hh, line 5 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsaWI9BOTM9U_JH0Aa:-MHPDDJ27NeQmpn4VyWm:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.hh#L5)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> #ifndef BTPrimaryGeneratorAction_h
> 
> #define BTPrimaryGeneratorAction_h 1
> ```
`#pragma once`
</blockquote></details>

done
___
*[BTPrimaryGeneratorAction.cpp, line 21 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGs_o_Z0QQzymmHFPuf:-MHPDFY4CIZT231ooMID:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.cpp#L21)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
>     fParticleGun(0),
> 
>     fEnvelopeBox(0)
> ```
same as above, prefer `nullptr` over `0`
</blockquote></details>

done
___
*[BTPrimaryGeneratorAction.cpp, line 79 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsaT0t40pcuYYtYq-n:-MHPDc-K7ig2D0dff-Ht:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.cpp#L79)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> G4double size = 0.8;
> 
>     G4double x0 = size * envSizeXY * (G4UniformRand() - 0.5);
> 
>     G4double yo = size * envSizeXY * (G4UniformRand() - 0.5);
> 
>     G4double z0 = -0.5 * envSizeZ;
> ```
these can all be `const`
</blockquote></details>

done


<!-- Sent from Reviewable.io -->

kylemtam reviewed (approved):
> Reviewed 1 of 2 files at r2, 6 of 9 files at r3.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1)* status: 7 of 10 files reviewed, 20 unresolved discussions (waiting on @dldyment, @jacobdeery, @kylemtam, @pauladam316, and @SaduniSumith)

*[a discussion](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiiunk-peBeax7hXc-:-MHiiunk-peBeax7hXc0:bmgg1dx) (no related file):*
Everything looks decent. I've added some comments with a few tiny requested changes. Pending addressing Jacob's few remaining comments, you've got my official approval.

---
___
*[BTDetectorConstruction.cpp, line 76 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZoQ18_90YYHuGcbx:-MHiglXw6FE5whvUgtwU:b-20ohfu) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L76)):*
<details><summary><i>Previously, dldyment (Delaney Dyment) wrote…</i></summary><blockquote>

this is another one of those things that G4 examples do that I don't completely understand. my assumption would be that it is intentional, but I'm afraid I don't have a solid answer rn. I'm looking into it. 
</blockquote></details>

I'd make note of this in your code with a comment - once you reach a point where you can actually run the program, you can try commenting out the new operation to see if it breaks anything
___
*[BTPrimaryGeneratorAction.cpp, line 33 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHihOiDDtb0A4fkg-Uh:-MHihOiDDtb0A4fkg-Ui:bbtbsm0) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTPrimaryGeneratorAction.cpp#L33)):*
> ```cpp
>     fParticleGun->SetParticleDefinition(particle);
>     fParticleGun->SetParticleMomentumDirection(G4ThreeVector(0., 0., 1.));
>     fParticleGun->SetParticleEnergy(18. * MeV);
> ```

any reason why the default particle energy is 18 MeV? I'm forgetting right now but for our application, would it be more appropriate to set it to closer to ~1-10 MeV?
___
*[BTRunAction.cpp, line 18 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiiPPB6xdjxv_AyPw8:-MHiiPPB6xdjxv_AyPw9:bbmvd7s) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L18)):*
> ```cpp
> 	fEdep2(0.)
> {
> 	///add more units to describe dose
> ```

do u know if Sv are built into the program?
___
*[BTRunAction.cpp, line 108 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiibM9AvxTQPNHg6ci:-MHiibM9AvxTQPNHg6cj:b-mmf5wq) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L108)):*
> ```cpp
>     fEdep += edep;
>     fEdep2 += edep * edep;
> }
> ```

Add an extra blank line just in case plz
___
*[BTSteppingAction.cpp, line 39 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiikqR4owWPhEb7qYj:-MHiikqR4owWPhEb7qYk:b-yjhrji) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.cpp#L39)):*
> ```cpp
> 	G4double edepStep = step->GetTotalEbergyDeposit();
> 	fEventAction->AddEdep(edepStep);
> }
> ```

add extra blank line plz


<!-- Sent from Reviewable.io -->

jacobdeery reviewed (changes requested):
Sorry for the hard block, but there's still at least one major comment that needs to be addressed before merging (the `#endif` statements), and I want to make sure it doesn't get accidentally merged since Kyle's approved it.

1. Ok. In general I prefer to see everything buildable and testable right from the beginning, but if this is part of the bare minimum required to build an application I guess we don't really have a choice.
2. Ok, that's fine for now - there are still a few inconsistencies, but not super important
3. Please do leave the oooo0000ooo out. It's distracting.

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-:-MHkzSeM9CftSx24VOBj:ba4wv3i)* status: 7 of 10 files reviewed, 26 unresolved discussions (waiting on @dldyment, @jacobdeery, @kylemtam, @pauladam316, and @SaduniSumith)

___
*[BTActionInitialzation.hh, line 19 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHkzjK1CgrXTAijuESF:-MHkzjK1CgrXTAijuESG:bfalicy) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTActionInitialzation.hh#L19)):*
> ```
> #endif
> ```
remove
___
*[BTDetectorConstruction.hh, line 5 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsYedDAzsdWlsTrNEr:-MHl-N7kAlBdWISbUCIH:bd7avmb) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.hh#L5)):*
<details><summary><i>Previously, dldyment (Delaney Dyment) wrote…</i></summary><blockquote>

done
</blockquote></details>

You'll also need to remove the #endif at the end of this file or it won't compile
___
*[BTDetectorConstruction.cpp, line 76 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZoQ18_90YYHuGcbx:-MHl-auP7fQoAeif23ND:b-bb1u4v) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L76)):*
<details><summary><i>Previously, kylemtam (Kyle Tam) wrote…</i></summary><blockquote>

I'd make note of this in your code with a comment - once you reach a point where you can actually run the program, you can try commenting out the new operation to see if it breaks anything
</blockquote></details>

Can you point me towards an example where G4 does this?
___
*[BTPrimaryGeneratorAction.hh, line 5 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsaWI9BOTM9U_JH0Aa:-MHl-gLaBarZBCqeo5j-:b-xzljqh) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.hh#L5)):*
<details><summary><i>Previously, dldyment (Delaney Dyment) wrote…</i></summary><blockquote>

done
</blockquote></details>

again, need to remove the endif
___
*[BTRunAction.h, line 28 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl-rr30PNDKJrLwE8A:-MHl-rr30PNDKJrLwE8B:bfalicy) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.h#L28)):*
> ```
> #endif
> ```
remove
___
*[BTRunAction.cpp, line 57 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl03n7-whCR-QRsGkx:-MHl03n7-whCR-QRsGky:b-2ot9i0) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L57)):*
> ```
>     G4double edep = fEdep.GetValue();
>     G4double edep2 = fEdep2.GetValue();
> ```
const
___
*[BTRunAction.cpp, line 60 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl01B74vV2n_GcDls9:-MHl01B74vV2n_GcDlsA:b-f48gqu) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L60)):*
> ```
>     G4double rms = edep2 - edep * edep / nofEvents;
>     if (rms > 0.) rms = std::sqrt(rms); else rms = 0.;
> ```
Two things here:
1. `rms` isn't a super descriptive name; what is this the RMS of?
2. Here you're reusing the `rms` variable. This is undesirable for two reasons: first, it means the name is inaccurate, since the value is only actually the RMS once you take the square root. Second, it means you can't make it `const`, which means that it can easily be accidentally changed further down. Prefer declaring two different variables instead.
___
*[BTRunAction.cpp, line 67 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0_pj3g6aMhxVPxsI:-MHl0_pj3g6aMhxVPxsJ:bevgwq7) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L67)):*
> ```
>     G4double mass = detectorConstruction->GetScoringVolume()->GetMass();
>     G4double dose = edep / mass;
>     G4double rmsDose = rms / mass;
> ```
consts
___
*[BTRunAction.cpp, line 69 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0fyEEbVxnuMYV25A:-MHl0fyEEbVxnuMYV25B:b3nmzel) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L69)):*
> ```
> BTPrimaryGeneratorAction*
> ```
you can use `auto` here
___
*[BTSteppingAction.h, line 13 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0m_P056u9UHGSZL6:-MHl0m_P056u9UHGSZL7:bvdxv5p) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.h#L13)):*
> ```objc
> {
> public:
> 	BTSteppingAction(BTEventAction* eventaction);
> ```

Please avoid mixing space and tab indentation
___
*[BTSteppingAction.h, line 23 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0lkU1YAzyQBjY22W:-MHl0lkU1YAzyQBjY22X:b-l6mwxj) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.h#L23)):*
> ```objc
> };
> 
> #endif
> ```

remove
___
*[BTSteppingAction.cpp, line 22 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0vV5BAWq1S0XD6o-:-MHl0vV5BAWq1S0XD6o0:b-ugqiui) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.cpp#L22)):*
> ```
> BTDetectorConstruction*
> ```
can be `auto`


<!-- Sent from Reviewable.io -->

dldyment committed:
Addressed comments

dldyment reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1)* status: 0 of 10 files reviewed, 26 unresolved discussions (waiting on @dldyment, @jacobdeery, @kylemtam, @pauladam316, and @SaduniSumith)

___
*[BTActionInitialzation.hh, line 19 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHkzjK1CgrXTAijuESF:-MJ2DRaABsntc0FxQ9Es:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTActionInitialzation.hh#L19)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> #endif
> ```
remove
</blockquote></details>

Done.
___
*[BTDetectorConstruction.hh, line 5 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsYedDAzsdWlsTrNEr:-MJ2DVBb3NQSdugHtwtX:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.hh#L5)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

You'll also need to remove the #endif at the end of this file or it won't compile
</blockquote></details>

done
___
*[BTDetectorConstruction.cpp, line 13 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsYr2n96298ERqNF9k:-MJ2DaWb0F-E3RC0_Jas:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L13)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> //....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
> ```
Were these just copied from the examples? I don't think we want them
</blockquote></details>

Done.
___
*[BTDetectorConstruction.cpp, line 23 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsYvmS5hDCLPbDlOg3:-MJ2DlkW49aF_fwjSBHn:b-4jve6) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L23)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> BTDetectorConstruction::~BTDetectorConstruction()
> 
> {}
> ```
I don't think we need to define the destructor if it doesn't do anything
</blockquote></details>

This is another of those things that G4 examples do that I don't really understand. 
___
*[BTDetectorConstruction.cpp, line 32 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZeJT0a-D1zwnvWib:-MJ2Dvl-0CGkdlvFYbFp:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L32)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

None of these change, right? Can we flag them as `const`?
</blockquote></details>

done
___
*[BTDetectorConstruction.cpp, line 76 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZoQ18_90YYHuGcbx:-MJ2E5ntA9ZmN2t4pkMV:b-3wj6bx) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L76)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

Can you point me towards an example where G4 does this?
</blockquote></details>

I can post the file in #payload that a lot of this is based off of, it does it for every geometry that it defines
___
*[BTPrimaryGeneratorAction.hh, line 5 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsaWI9BOTM9U_JH0Aa:-MJ2EJH8-pYixpVb1Nkl:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.hh#L5)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

again, need to remove the endif
</blockquote></details>

done
___
*[BTPrimaryGeneratorAction.cpp, line 74 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsaByO2BiD7uF5G2PO:-MJ2EpZ890OSHZqrXky3:bb7f2mf) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.cpp#L74)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

The logic here can be simplified by moving the code from the `if (!fEnvelopeBox)` block into this `else` block
</blockquote></details>

done
___
*[BTPrimaryGeneratorAction.cpp, line 76 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGscHFT-lqoU-QjOcTv:-MJ2EqcV9nD9KtpDNI7n:b-kekorm) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTPrimaryGeneratorAction.cpp#L76)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> G4double size = 0.8;
> ```
Why do we scale by 0.8?
</blockquote></details>

we don't need to, I removed it
___
*[BTPrimaryGeneratorAction.cpp, line 33 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHihOiDDtb0A4fkg-Uh:-MJ2EM875Y-gTqhgoh2L:b-jtqmb2) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTPrimaryGeneratorAction.cpp#L33)):*
<details><summary><i>Previously, kylemtam (Kyle Tam) wrote…</i></summary><blockquote>

any reason why the default particle energy is 18 MeV? I'm forgetting right now but for our application, would it be more appropriate to set it to closer to ~1-10 MeV?
</blockquote></details>

the build test isn't really similar to the flight scenario we're building after, it's a lot more basic. Cosmic rays will need to be modeled by a number of different particles/energies, but I wanted something less complicated for this. I chose 18MeV because that's one of the energies I tested on the linac at work that day, I just needed a reasonable number. 
___
*[BTRunAction.h, line 28 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl-rr30PNDKJrLwE8A:-MJ2EtpoBZnnWOmyZm1A:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.h#L28)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> #endif
> ```
remove
</blockquote></details>

Done.
___
*[BTRunAction.cpp, line 18 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiiPPB6xdjxv_AyPw8:-MJ2Eub28wBRaSbe0qZa:b-7b1kme) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L18)):*
<details><summary><i>Previously, kylemtam (Kyle Tam) wrote…</i></summary><blockquote>

do u know if Sv are built into the program?
</blockquote></details>

Probably? If not we could just convert them in the program. All dose measurement types are connected really conveniently. 
___
*[BTRunAction.cpp, line 57 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl03n7-whCR-QRsGkx:-MJ2F7h_EP8vq9-sL1b-:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L57)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
>     G4double edep = fEdep.GetValue();
>     G4double edep2 = fEdep2.GetValue();
> ```
const
</blockquote></details>

Done.
___
*[BTRunAction.cpp, line 60 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl01B74vV2n_GcDls9:-MJ2F91B9RI0onq7Ht2k:b-2p3rma) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L60)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
>     G4double rms = edep2 - edep * edep / nofEvents;
>     if (rms > 0.) rms = std::sqrt(rms); else rms = 0.;
> ```
Two things here:
1. `rms` isn't a super descriptive name; what is this the RMS of?
2. Here you're reusing the `rms` variable. This is undesirable for two reasons: first, it means the name is inaccurate, since the value is only actually the RMS once you take the square root. Second, it means you can't make it `const`, which means that it can easily be accidentally changed further down. Prefer declaring two different variables instead.
</blockquote></details>

1. in this case edep stands for "energy deposited", which rms is the root mean square of. edep is a measurement that G4 takes and is used to calculate dose, since dose = energy deposited / mass. the rms of edep (which I've now changed from rms to rmsEdep) is used to calculate the variance in dose (rmsEdep / mass). 
___
*[BTRunAction.cpp, line 67 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0_pj3g6aMhxVPxsI:-MJ2G-Ib36JEvxoAFHRj:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L67)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
>     G4double mass = detectorConstruction->GetScoringVolume()->GetMass();
>     G4double dose = edep / mass;
>     G4double rmsDose = rms / mass;
> ```
consts
</blockquote></details>

Done.
___
*[BTRunAction.cpp, line 69 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0fyEEbVxnuMYV25A:-MJ2G-vy8ZCZAM0mdPJz:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L69)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> BTPrimaryGeneratorAction*
> ```
you can use `auto` here
</blockquote></details>

Done.
___
*[BTRunAction.cpp, line 108 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiibM9AvxTQPNHg6ci:-MJ2G0aI0K2hXzu-f5LN:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L108)):*
<details><summary><i>Previously, kylemtam (Kyle Tam) wrote…</i></summary><blockquote>

Add an extra blank line just in case plz
</blockquote></details>

Done.
___
*[BTSteppingAction.h, line 13 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0m_P056u9UHGSZL6:-MJ2G1C51EH0xCYY2lB3:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.h#L13)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

Please avoid mixing space and tab indentation
</blockquote></details>

Done.
___
*[BTSteppingAction.h, line 23 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0lkU1YAzyQBjY22W:-MJ2G20c2c1pB8CkyDMU:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.h#L23)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

remove
</blockquote></details>

Done.
___
*[BTSteppingAction.cpp, line 22 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0vV5BAWq1S0XD6o-:-MJ2G2ay7OxvLziyayAw:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.cpp#L22)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> BTDetectorConstruction*
> ```
can be `auto`
</blockquote></details>

Done.
___
*[BTSteppingAction.cpp, line 39 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiikqR4owWPhEb7qYj:-MJ2G3B694Q4Gwx7kcKL:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.cpp#L39)):*
<details><summary><i>Previously, kylemtam (Kyle Tam) wrote…</i></summary><blockquote>

add extra blank line plz
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

dldyment reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1)* status: 0 of 10 files reviewed, 26 unresolved discussions (waiting on @dldyment, @jacobdeery, @kylemtam, @pauladam316, and @SaduniSumith)

___
*[BTRunAction.cpp, line 60 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl01B74vV2n_GcDls9:-MJ2GO02DwL6rWr54xft:baeg75t) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L60)):*
<details><summary><i>Previously, dldyment (Delaney Dyment) wrote…</i></summary><blockquote>

1. in this case edep stands for "energy deposited", which rms is the root mean square of. edep is a measurement that G4 takes and is used to calculate dose, since dose = energy deposited / mass. the rms of edep (which I've now changed from rms to rmsEdep) is used to calculate the variance in dose (rmsEdep / mass). 
</blockquote></details>

2. done


<!-- Sent from Reviewable.io -->

jacobdeery reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1)* status: 0 of 10 files reviewed, 13 unresolved discussions (waiting on @dldyment, @jacobdeery, @kylemtam, @pauladam316, and @SaduniSumith)

___
*[BTRunAction.cpp, line 69 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0fyEEbVxnuMYV25A:-MJEjCDnB4sbhkgnFscu:b7w2jv3) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L69)):*
> ```
> auto BTPrimaryGeneratorAction* generatorAction
> ```
should just be `auto generatorAction`; `auto` tells the compiler to deduce the type, so you don't need to list it again
___
*[BTRunAction.cpp, line 108 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiibM9AvxTQPNHg6ci:-MJEjMTe6wz0uY4QEczz:b-g1e0an) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L108)):*
<details><summary><i>Previously, dldyment (Delaney Dyment) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

in case what?
___
*[BTSteppingAction.cpp, line 22 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0vV5BAWq1S0XD6o-:-MJEjKoj78Ln8L8-BouL:bnj7hl0) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.cpp#L22)):*
<details><summary><i>Previously, dldyment (Delaney Dyment) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

same as above


<!-- Sent from Reviewable.io -->

kylemtam committed:
Fix variale type inferences

Addressed two of Jacob's comments

kylemtam reviewed (commented):
> Reviewed 10 of 10 files at r4, 2 of 2 files at r5.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1)* status: all files reviewed, 10 unresolved discussions (waiting on @dldyment, @jacobdeery, @pauladam316, and @SaduniSumith)

___
*[BTRunAction.cpp, line 69 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0fyEEbVxnuMYV25A:-MJPrZ9n5x1NOe7_hDzL:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L69)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> auto BTPrimaryGeneratorAction* generatorAction
> ```
should just be `auto generatorAction`; `auto` tells the compiler to deduce the type, so you don't need to list it again
</blockquote></details>

Done.
___
*[BTRunAction.cpp, line 108 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiibM9AvxTQPNHg6ci:-MJPqa0WBAehUZcAhVV1:blr8l25) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L108)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

in case what?
</blockquote></details>

Isn't it a general practice to always leave a newline at the end of a file (ignore 'just in case')? (<-- Not sure the reason for the newline tho, could you explain this Jacob?)
___
*[BTSteppingAction.cpp, line 22 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHl0vV5BAWq1S0XD6o-:-MJPs5WR8G1b3hDOTld9:b-896fix) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTSteppingAction.cpp#L22)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

same as above
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

jacobdeery reviewed (approved):
There are still a few formatting issues that I think should be addressed, but I'm approving because I know you don't have tooling for autoformatting currently. Will open an issue for this.

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-:-MJPugmE-EGWV2so6x5G:bni59dl)* status: all files reviewed, 7 unresolved discussions (waiting on @dldyment, @jacobdeery, @kylemtam, @pauladam316, and @SaduniSumith)

___
*[BTDetectorConstruction.cpp, line 76 at r1](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MGsZoQ18_90YYHuGcbx:-MJPCU6PBUeDOwREsQzM:b-6p8f70) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/a53cc80bf95158aea37fe9878035d15651991e55/BTDetectorConstruction.cpp#L76)):*
<details><summary><i>Previously, dldyment (Delaney Dyment) wrote…</i></summary><blockquote>

I can post the file in #payload that a lot of this is based off of, it does it for every geometry that it defines
</blockquote></details>

Ok. Weird.
___
*[BTRunAction.cpp, line 108 at r3](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MHiibM9AvxTQPNHg6ci:-MJPuQEJATsITuofBUZj:bo26hln) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/5d55931b415f9da36c064cbfc6ea95a76b3fd7f2/BTRunAction.cpp#L108)):*
<details><summary><i>Previously, kylemtam (Kyle Tam) wrote…</i></summary><blockquote>

Isn't it a general practice to always leave a newline at the end of a file (ignore 'just in case')? (<-- Not sure the reason for the newline tho, could you explain this Jacob?)
</blockquote></details>

Yep; it's mostly just a style thing. I was curious what the "just in case" was for.
___
*[BTSteppingAction.cpp, line 37 at r5](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MJPue0k28F1HuOT22wp:-MJPue0lAPxw29pqgGNT:b-za4fyy) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/f927cb4c1a8e7027f391ea5a02787e0ed94ff57b/BTSteppingAction.cpp#L37)):*
> ```
> Ebergy
> ```
Should this be `Energy`?


<!-- Sent from Reviewable.io -->

kylemtam committed:
Fix 'Ebergy' typo

kylemtam reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1)* status: 9 of 10 files reviewed, 6 unresolved discussions (waiting on @dldyment, @jacobdeery, @kylemtam, @pauladam316, and @SaduniSumith)

___
*[BTSteppingAction.cpp, line 37 at r5](https://reviewable.io/reviews/waterloo-rocketry/payload-sims/1#-MJPue0k28F1HuOT22wp:-MJPwc1X3n0OxupNlhjo:b-56foht) ([raw file](https://github.com/waterloo-rocketry/payload-sims/blob/f927cb4c1a8e7027f391ea5a02787e0ed94ff57b/BTSteppingAction.cpp#L37)):*
<details><summary><i>Previously, jacobdeery (Jacob Deery) wrote…</i></summary><blockquote>

> ```
> Ebergy
> ```
Should this be `Energy`?
</blockquote></details>

Just checked the documentation. This should be `Energy` and has been corrected.


<!-- Sent from Reviewable.io -->
